src/core/ActivityInfoTree.cpp

```diff
@@ -283,5 +285,66 @@ void ActivityInfoTree::init()
 
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
+
+
+}
+void ActivityInfoTree::display(const QString& text)
+{
+
+    if(text.trimmed().length() < m_searchedtext.length() or text.trimmed() == ""){
+        if(text.trimmed() == ""){
+            m_searchedtext = "";
+            m_searched = m_menuTreeFull;
+        }
+
+        m_menuTree = m_menuTreeFull;
+    }
+
+
+
+    if(!text.trimmed().isEmpty() and text.length() != m_searchedtext.length())
+        m_menuTree.erase(std::remove_if(m_menuTree.begin(),m_menuTree.end(),
```

Comment:
as discussed in the other PR, it would be great to be able to also search on the internal name, can you add it?